tests/integration/api/test_activation_with_credential.py

```diff
@@ -346,3 +445,113 @@ def test_create_activation_with_conflict_credentials(
         "in extra var. It conflicts with credential type: type1. "
         "Please check injectors." in response.data["non_field_errors"]
     )
+
+
+@pytest.mark.django_db
+def test_create_activation_without_extra_vars(
+    client: APIClient,
+    kafka_credential_type: models.CredentialType,
+    preseed_credential_types,
+):
+    fks = create_activation_related_data(OVERLAP_EXTRA_VAR)
+    test_activation = TEST_ACTIVATION.copy()
+    test_activation.pop("extra_var_id")
+    test_activation["decision_environment_id"] = fks["decision_environment_id"]
+    test_activation["rulebook_id"] = fks["rulebook_id"]
+
+    credential_types = models.CredentialType.objects.bulk_create(
+        [
+            models.CredentialType(
+                name="user_type",
+                inputs={"fields": [{"a": "b"}]},
+                injectors={"extra_vars": {"username": "adam"}},
+            ),
+            models.CredentialType(
+                name="password_type",
+                inputs={"fields": [{"1": "2"}]},
+                injectors={"extra_vars": {"password": "secret"}},
+            ),
+        ]
+    )
+
+    eda_credentials = models.EdaCredential.objects.bulk_create(
+        [
+            models.EdaCredential(
+                name="credential-1",
+                inputs={"sasl_username": "adam", "sasl_password": "secret"},
+                credential_type_id=credential_types[0].id,
+            ),
+            models.EdaCredential(
+                name="credential-2",
+                inputs={"sasl_username": "bearny", "sasl_password": "demo"},
+                credential_type_id=credential_types[1].id,
+            ),
+        ]
+    )
+
+    eda_credential_ids = [credential.id for credential in eda_credentials]
+    test_activation["eda_credentials"] = eda_credential_ids
+
+    response = client.post(f"{api_url_v1}/activations/", data=test_activation)
+    assert response.status_code == status.HTTP_201_CREATED
```

Comment:
@hsong-rh since we are using 2 credentials with the same values but different injectors a cleaner test would be to convert this to a failure because of extra var name conflicts.
And create another one with just 1 credential with no conflict

Reply:
@mkanoor Such test cases have already covered in `test_create_activation_with_key_conflict` and `test_create_activation_with_compatible_credentials`